lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php

```diff
@@ -420,6 +406,49 @@ private function createRecordQueryParams($ESQuery, SearchEngineOptions $options,
         return $params;
     }
 
+    private function getAggregationQueryParams(SearchEngineOptions $options, RecordHelper $recordHelper)
+    {
+        $acl = null;
+        if ($this->app['authentication']->isAuthenticated()) {
+            $acl = $this->app['acl']->get($this->app['authentication']->getUser());
+        }
+
+        $rights = [];
+        foreach ($options->getDataboxes() as $databox) {
+            $can_see_business = (null !== $acl && $acl->can_see_business_fields($databox));
+            $rights[$databox->get_sbas_id()] = $can_see_business;
+        }
+
+        $params = [];
+        foreach ($recordHelper->getFieldsStructure() as $name => $field) {
+            if (!$field['searchable'] || !$field['to_aggregate']) {
+                continue;
+            }
+            $params[$name] = [
```

Comment:
I can see Gizeh pyramids here, maybe something like `$query['bool']['should']['...']` will be better.